Sources/tart/OCI/Registry.swift

```diff
@@ -8,11 +8,27 @@ enum RegistryError: Error {
 }
 
 struct TokenResponse: Decodable {
-  let creationTime = Date()
-  
   var token: String
-  var expires_in: Int?
-  
+  var expiresIn: Int?
+  var issuedAt: Date = Date()
+
+  enum CodingKeys: String, CodingKey {
```

Comment:
I'm not a big fan of writing custom decoding that way. `TokenResponse` is too simple for that. Let's maybe use [`dateDecodingStrategy`](https://developer.apple.com/documentation/foundation/jsondecoder/2895216-datedecodingstrategy) of the `JSONDecoder`?

Reply:
I don't think this is a good idea:

1. Using snake case might simplify things at first, but once we start using a code formatter that enforces [official Swift's coding conventions](https://www.swift.org/documentation/api-design-guidelines/#general-conventions) these fields case will be converted and token parsing will break suddenly
2. Leaking abstractions: to parse the `TokenResponse` correctly we will need a specifically-configured `JSONDecoder` instance, so naturally we may want to hide these details in a custom `TokenResponse.init()`, however from the outside view the `TokenResponse` will still be `Decodable`, and thus one might decode it using some other `JSONDecoder` instance accidentally
3. General awkwardness: since there's a `let` field in `TokenResponse`, we cannot simply assign `JSONDecoder().decode(...)` result to `self` in custom `init()`, we will need to assign each `var` field one by one

Reply:
1. There is `decoder.keyDecodingStrategy = .convertFromSnakeCase`. 🤔 
2. Might use a static method like `TokenResponse.parse`.
3. I think it's less awkward then having the keys and custom decoding logic. We can make `TokenResponse` `fileprivate` so it's not leaking beside the file.

Reply:
See 7a53744d5b7205b474c0cb908569e9d3adda5f1a.